core/modules/wsched/ScanScheduler.cc

```diff
@@ -220,6 +220,46 @@ util::Command::Ptr ScanScheduler::getCmd(bool wait)  {
 }
 
 
+void ScanScheduler::queCmd(util::Command::Ptr const& cmd) {
+    std::vector<util::Command::Ptr> vect;
+    vect.push_back(cmd);
+    queCmd(vect);
+}
+
+
+void ScanScheduler::queCmd(std::vector<util::Command::Ptr> const& cmds) {
+
+    {
+        std::lock_guard<std::mutex> lock(util::CommandQueue::_mx);
+
+        for (auto const& cmd:cmds) {
+            wbase::Task::Ptr t = std::dynamic_pointer_cast<wbase::Task>(cmd);
+            if (t == nullptr) {
+                LOGS(_log, LOG_LVL_WARN, getName() << " queCmd could not be converted to Task or was nullptr");
```

Comment:
This looks scarier to me than just a warning. Why not to throw an exception right away as this could be a logic problem within the code?

Reply:
There are valid commands that are not of type Task. They are usually only seen when the program is shutting down.